src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -369,6 +369,19 @@ def get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood=None, release
         raise CLIInternalError("No content was found in helm registry path response.")
 
 
+def send_request_with_retries(cli_ctx, method, url, headers, fault_type, summary, uri_parameters=None, resource=None, retry_count=5, retry_delay=3):
+    for i in range(retry_count):
+        try:
+            response = send_raw_request(cli_ctx, method, url, headers=headers, uri_parameters=uri_parameters, resource=resource)
+            if response.status_code == 200:
```

Comment:
this code wont return anything(and also not error out) if the API returns non-200 status code after all retries. We should return the response in the last try even if it is not 200 and then do the handling for non-200 code in the caller of this method.